src/platform/packages/shared/kbn-zod-helpers/src/date_from_string.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import * as z from '@kbn/zod';
+
+export const dateFromString = z
```

Comment:
Why was it required to add a custom schema? Zod's [datetime](https://zod.dev/?id=datetimes) seems doing the same.

Reply:
z.string().datetime() validates the string as a valid date string. It does not transform it to a Date which I wanted

But actually I could use `z.coerce.date()` but somehow I thought we were using an older version of zod which did not have it: https://zod.dev/?id=datetimes#:~:text=%7D)%3B-,Coercion%20to%20Date,-Since%20zod%203.20

Reply:
Actually... it does not work with our router... I remember now why i had to create my own type...
![image](https://github.com/user-attachments/assets/3781da00-a740-4054-bd30-a5a392c90016)